### PR TITLE
adds tests for timestamps, escaped characters, fieldnames and annotations of length 13 and 14, a test for a struct in which the same fieldname is used multiple times, and a test to verify struct field digests are sorted first, then escaped

### PR DESCRIPTION
**Pull request by pbcornell**
Various Ion Hash tests accumulated during development of python and js implementations.

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: ion_hash_tests.ion [hunk 1]
```diff
@@ -899,6 +919,39 @@ sexpCount128::{
...
+    update::(0x0b) update::(0xd0)
+        update::(0x0b) update::(0x70) update::(0x68 0x65 0x6c 0x6c 0x6f 0x68 0x65 0x6c 0x6c 0x6f 0x68 0x65 0x6c) update::(0x0e)
+        update::(0x0b) update::(0x20) update::(0x01) update::(0x0e)
+        digest::(0x0b 0x70 0x68 0x65 0x6c 0x6c 0x6f 0x68 0x65 0x6c 0x6c 0x6f 0x68 0x65 0x6c 0x0e 0x0b 0x20 0x01 0x0e)
+      update::(0x0c 0x0b 0x70 0x68 0x65 0x6c 0x6c 0x6f 0x68 0x65 0x6c 0x6c 0x6f 0x68 0x65 0x6c 0x0c 0x0e 0x0c 0x0b 0x20 0x01 0x0c 0x0e)
+    update::(0x0e)
+    digest::(0x0b 0xd0 0x0c 0x0b 0x70 0x68 0x65 0x6c 0x6c 0x6f 0x68 0x65 0x6c 0x6c 0x6f 0x68 0x65 0x6c 0x0c 0x0e 0x0c 0x0b 0x20 0x01 0x0c 0x0e 0x0e)
+  ) }
+}
+{ ion:{hellohellohell: 1},
```
[the remark this answers is not in the document]
Comment (pbcornell) [reply ion_hash_tests.ion]: These two assert correct behavior at the boundary between annotations of length 13 and 14.  In the previous test, the binary writer produces `L=13` and no `length` subfield, whereas in this test the binary writer produces `L=14` and a `length` subfield.  In both cases, we want to confirm the hash impl changes `L` (aka `Q`, in the context of the Ion Hash spec) to `0`, and does not output any bits from a `length` subfield, if present.

File: ion_hash_tests.ion [hunk 2]
```diff
@@ -928,6 +981,35 @@ sexpCount128::{
              0x0e)
   ) }
 }
+{
+  ion:{a:3, a:1, a:2},
+  expect:{ identity:(
+    update::(0x0b) update::(0xd0)
+        update::(0x0b) update::(0x70) update::(0x61) update::(0x0e)  // c
```
[the remark this answers is not in the document]
Comment (pbcornell) [reply ion_hash_tests.ion]: fixed, thank you

File: ion_hash_tests.ion [hunk 3]
```diff
@@ -928,6 +981,35 @@ sexpCount128::{
...
+        update::(0x0b) update::(0x20) update::(0x03) update::(0x0e)  // 3
+        digest::(0x0b 0x70 0x61 0x0e 0x0b 0x20 0x03 0x0e)
+
+        update::(0x0b) update::(0x70) update::(0x61) update::(0x0e)  // a
+        update::(0x0b) update::(0x20) update::(0x01) update::(0x0e)  // 1
+        digest::(0x0b 0x70 0x61 0x0e 0x0b 0x20 0x01 0x0e)
+
+        update::(0x0b) update::(0x70) update::(0x61) update::(0x0e)  // b
```
[the remark this answers is not in the document]
Comment (pbcornell) [reply ion_hash_tests.ion]: fixed, thank you

File: ion_hash_tests.ion (hunk 1 above)
[the remark this answers is not in the document]
Comment (tgregg) [reply ion_hash_tests.ion]: These are field names though, not annotations. What am I missing?

[the remark this answers is not in the document]
Comment (pbcornell) [reply ion_hash_tests.ion]: The confusion is mine--replace "annotation" with "field name" in my previous comment.

Also realized we're missing similar boundary tests for annotations, so I've added them.

**Pull request merged**